### DISCUSSION TITLE
Next SQLite release looks like it will include more math functions

### DISCUSSION THREAD

**Discussion by justinclift** [sqlitebrowser/sqlitebrowser#2507] [General]
Just noticed the development code for the next SQLite release is now including more math functions: https://github.com/sqlite/sqlite/commit/63f8f98a63ef23803f663e6b128d9b3fe8ef53b6

That's likely to be useful for people. :smile:

**Comment by esrat** [comment]
Hi,
I am also exited about the new release ... but not in terms of new functionality, but in being able to browse my data again! ;-P
Until a few days ago my table only had around 65,000,000 records and I could browse, filter, sort it. I never realized that the "Prefetch block size" setting was completely ignored, because I got a pretty beefy computer.
But now my dataset grew above 95,000,000 records - rendering the "Browse Data" view completely useless!
This seems to be fixed already in the nightly builds. ... I cannot wait to find the next release!
(I am using the portable version which, of course, is only available for releases.)

**Comment by chrisjlocke** [comment]
>  I never realized that the "Prefetch block size" setting was completely ignored

It's not is it?  Where did you pick that up from?

> I am using the portable version which, of course, is only available for releases.

Technically any nightly is 'portable' as to run it all you do is extract the .zip file.... assuming you're on Windows, of course...

**Reply by esrat** [reply]
> It's not is it? Where did you pick that up from?

For the most current nightly build I see something like "1 - 37 of 49999" which is because of the default setting of a prefetch block size of 50000. For the last release I only saw statements like "1 - 16 of 65604207", so I thought, it did load all of it, always. But maybe the whole table was in RAM by this time, so I saw only the end result. - But with the new table this does not work anymore. The last thing I see is "0 - 94802510 of 9480251" (not enough space to show the whole string) - and then I have to kill the process, because it makes no sense to wait. The browser gets stuck in some endless loop.

> Technically any nightly is 'portable' as to run it all you do is extract the .zip file.... assuming you're on Windows, of course...

Yes, for my new tests I replaced the contents of the App\SQLiteDatabaseBrowser64\ directory of the portable installation with the contents of the .ZIP archive. Now I can load the "Browse data" tab again. - But sadly it has still more bugs, so that I cannot use it like mentioned before. I was too enthusiastic when I saw these first lines of data filling in. :-(
